### ISSUE TITLE
DHT not get the values

### ISSUE THREAD

**Issue by edgarreis** [jescarri/nodemcu-dht22-mqtt#1]
Hi, I am trying connect DHT22 on NodeMCU also, but this code github give me the error that cannot get the values of sensor and return nil. Even I following all steps. Did have some problem like that?

**Comment by jescarri** [comment]
Hey, on what Pin are you connecting the DHT?

I'm not using anymore this code, I switched to a C++ code base, Lua and the NodeMCU firmware where having a lot of problems.

Try this new CodeBase
https://github.com/jescarri/nodemcu-elastic-search

Is using arduino and coreos etcd for dynamic configuration but you can remove that part.

Let me know how it goes, I will be updating the documentation in the upcoming days.

**Comment by jescarri** [comment]
Hey did you make it work?.
I will close this issue re-open it if you have any problem.

**Comment by edgarreis** [comment]
Hi!

Unfortunately the lua code doesn't works.
But .ino does. ;)

I'm trying others ways in lua code because all my project is written in lua.

As you told lua has a lot of problems. --'

Are you still programming the Esp8266?
Maybe we could talk in Skype someday..

Get Outlook for Android<https://aka.ms/ghei36>

**Comment by jescarri** [comment]
Yes I still program the ESP8266 from time to time, right now I'm focussing on learning go.

Last time I used this firmware I switched to the latest float firmware.

beware that you need to connect the DHT vcc pin to gpio2 and DHT Data to gpio3.

what error are you getting?, can you paste the debug output?

**Comment by edgarreis** [comment]
>NodeMCU 0.9.*** You've got 3 sec to stop timer 0 ***
 IP192.168.15.5
MAC18-FE-34-C1-1D-4A
Hurray! GOT NETWORK!
PANIC: unprotected error in call to Lua API (**read_dht.lua:7: attempt to index global 'dht' (a nil value))**
PANIC: unprotected error in call to Lua API (attempt to index a nil value)
PANIC: unprotected error in call to Lua API (attempt to index a nil value)
............. PANIC until reboot


PS:
-only to make sure I used your .bin firmare
-yes, the pins that I used is the same that you told me.

**Comment by jescarri** [comment]
Hmmm let me test it today once i get home, i have the .bin somewhere, maybe
I didn't uploaded it to the repo, and in older versions nodemcu didn't had
support for dht.
